### DISCUSSION TITLE
Has 'capacitor' been considered for mobile app development

### DISCUSSION THREAD

**Discussion by camstuart** [reflex-dev/reflex#3152] [Ideas]
It would seem that [capacitor](https://capacitorjs.com/) is a great way to take a web app's ui to a 'native' (sort of) mobile app. 

I don't think this would be especially hard. As long as the front end JS can be passed into capacitor, and told what base url the backend is on, it should be easy.

One step further would be to wrap the [ionic framework](https://ionicframework.com/) widget react components, which I beleive is the magic of reflex.

Has anyone spoken / considered this?

While it's a bit of a slog, I would be happy to work on these things. Building ionic/capacitor apps in python reflex would be just awesome.

But to be truly awesome, PWA capabilities would be important, the mobile world is all about 'offline first'

**Comment by picklelo** [comment]
@camstuart We've had many requests to support mobile apps but our core team doesn't have bandwidth to take this on. We would happily support anyone who does want to work on this project - I agree that it should be possible technically and would be pretty awesome to extend to mobile and desktop apps.

**Reply by s4n-cz** [reply]
I was just checking if creating a PWA is possible in Reflex. This would be a really nice feature.

**Reply by masenf** [reply]
PWA is not currently possible, but we do want to deliver some PWA-enabling features in the next few months, like client-only state and client-only event handlers, which should improve the situation.

Some of this is on the [roadmap](https://github.com/reflex-dev/reflex/issues/2727) as "Decentralized Var Operations", and a lot of it is not fully formulated yet.

From my perspective, PWA is more valuable than native app support for reflex, even though PWA support is still not on par with the native experience, i think it ultimately will win out once the compatibility issues are fixed.
